src/ec/packages/ec-component-ecl-compliance/components/ecl-compliance-media-container.html.twig

```diff
@@ -0,0 +1,56 @@
+{#
+  Parameters:
+    alt (string),
+    image (string),
+    embedded_media: (string),
+    sources: (array of objects),
+    tracks: (array of objects)
+#}
+
+{% extends "@ecl-twig/ec-component-ecl-compliance/ecl-compliance.html.twig" %}
+
+{% set message = '' %}
+{% set component = 'media-container' %}
+{% set no_embedded_media_case = embedded_media is not defined or embedded_media is empty %}
+{% set no_sources_case = sources is not defined or sources is empty or sources is not iterable %}
+{% set no_tracks_case = tracks is not defined or tracks is empty or tracks is not iterable %}
+{% set no_video_resources_case = no_sources_case and no_tracks_case %}
+{% set no_image_case = image is not defined or image is empty %}
+{% set no_alt_case = no_image_case == false and (alt is not defined or alt is empty) %}
+
+{# no media provided at all #}
+{% set not_compliant = no_image_case and no_embedded_media_case and no_video_resources_case %}
+{% if not_compliant %}
+  {% set message = 'One media is required for the media-container component, an "image", an "embedded_media" or video "tracks" and "sources".' %}
+{% endif %}
+{# No embedded_media, we check for video sources #}
+{% if no_embedded_media_case and no_sources_case == false %}
+  {% for source in sources %}
+    {% if (source.src is not defined or source.src is empty) or (source.type is not defined or source.type is empty) %}
+      {% set not_compliant = true %}
+    {% endif %}
+  {% endfor %}
+  {% if not_compliant %}
+    {% set message = message ~ 'The "sources" parameter should be an array with a "src" and a "type" for each element.' %}
+  {% endif %}
+{% endif %}
+{# No embedded_media, we check for video tracks #}
+{% if no_embedded_media_case and no_tracks_case == false %}
+  {% for track in tracks %}
+    {% if (track.kind is not defined or track.kind is empty) or
+      (track.label is not defined or track.label is empty) or
+      (track.src is not defined or track.src is empty) or
+      (track.src_lang is not defined or track.src_lang is empty) %}
+      {% set not_compliant = true %}
+      {% set tracks_issue = true %}
+    {% endif %}
+  {% endfor %}
+  {% if not_compliant and tracks_issue is defined %}
```

Comment:
Similar comment as in https://github.com/ec-europa/ecl-twig/pull/423/files#r421329856
Changing the `not_compliant` marker inside a loop allows you for adding more specific information which iteratee is wrong exactly.
It's also not clear to me why there's the additional `tracks_issue` here when it's not used for providing additional context.

Reply:
`tracks_issue` is there because `not_compliant` might be true from the previous loop (sources), therefore we need and additional flag to understand whether to add the messages about the tracks or not.

Reply:
clearly you can have different variables for the separate complience checks then instead of working around this.